src/Database/LSMTree/Internal/MergingRun.hs

```diff
@@ -781,6 +783,34 @@ remainingMergeDebt (DeRef mr) = do
         let debt = totalDebt - (spent + unspent)
         return (MergeDebt debt, size)
 
+{-# INLINE supplyChecked #-}
+-- | Helper function to assert common invariants for functions that supply
+-- credits.
+supplyChecked ::
+     forall m r s. (HasCallStack, Monad m)
```

Comment:
This constraint is redundant when `!NO_IGNORE_ASSERTS` and generates a warning.

Reply:
Yeah, fixed now, but it's a recurring annoying issue.